lib/Sema/SemaBounds.cpp

```diff
@@ -2145,11 +2153,143 @@ namespace {
 
 
   public:
-    CheckBoundsDeclarations(Sema &S, BoundsExpr *ReturnBounds) : S(S),
+    CheckBoundsDeclarations(Sema &S, Stmt *Body, CFG *Cfg, BoundsExpr *ReturnBounds) : S(S),
       DumpBounds(S.getLangOpts().DumpInferredBounds),
       PointerWidth(S.Context.getTargetInfo().getPointerWidth(0)),
+      Body(Body),
+      Cfg(Cfg),
       ReturnBounds(ReturnBounds) {}
 
+    typedef llvm::SmallPtrSet<const Stmt *, 16> StmtSet;
+
+    void IdentifyChecked(Stmt *S, StmtSet &CheckedStmts, bool InCheckedScope) {
+      if (!S)
+        return;
+
+      if (InCheckedScope)
+        if (isa<Expr>(S) || isa<DeclStmt>(S) || isa<ReturnStmt>(S))
+          CheckedStmts.insert(S);
+
+      if (CompoundStmt *CS = dyn_cast<CompoundStmt>(S))
```

Comment:
LLVM recommends adding "const" when using dyc_cast for type testing
if (const type *Val = dyn_cast<T>(oldVal))